### ISSUE TITLE
How to bind username with the user's inputs ?

### ISSUE THREAD

**Issue by younessadmi** [ldaptools/ldaptools-bundle#17]
Hello,

First, i'd like to thank for this bundle. LDAP Authentication is killing my mind since long time.

Is it possible to avoid to provide the `ldap_tools.domains.example.username`  in `app/config/config.yml` ?

I want the user to provide his own username and password to bind the ldap server

**Comment by younessadmi** [comment]
Hi there,

I finally found a solution randomly. By leaving the username and password empty, it works ! :+1: 

```
ldap_tools:
    domains:
        company:
            domain_name: vnet.company.com
            username: 
            password: 
            base_dn: "dc=vnet,dc=company,dc=com"
            servers: ['ldap01.vnet.company.com']
```

**Comment by ChadSikorra** [comment]
Currently this is not possible, but it should be. The problem at the moment is I'm requiring the username/password in the config tree:

https://github.com/ldaptools/ldaptools-bundle/blob/master/DependencyInjection/Configuration.php#L133

So I'd have to remove that requirement. Easy enough. But then I also need to adjust the Guard and LdapAuthenticationProvider to attempt a bind with user credentials prior to searching for the user.

**Edit**: You got it working as I was typing this, hah. You must be using a user provider that's not LDAP? I'm going to re-open as this is still a feature I think would be handy for those wanting to use the LDAP user provider. And you shouldn't have to provide an empty username/password.

**Comment by Xeyos88** [comment]
I have the same problem, I would like to use this configuration to log in and then retrieve the data from the DB.
But it does not seem possible to get username and password from the login form.
I need to add the bind_format uid=username too...

**Comment by ChadSikorra** [comment]
@Xeyos88 I'm not sure that's the same problem? This issue this is about removing the need to enter a username/password in the LdapTools domain config in `config.yml`, and instead just rely on the username/password for both the LDAP user provider search and then the bind for the authentication provider.

Unless you're talking about using the user supplied username/password from the login form and then using those some credentials for subsequent LDAP calls after they login? This is possible, but you need to utilize the events properly to store the user credentials so you can use it later.

Or maybe I'm just misunderstanding your request?

Anywho, I should be able to get around to some improvements soon. Busy with work and moving apartments lately. Should have more free time soon.

**Comment by Xeyos88** [comment]
Yes, what I mean is just based on the username and password of the user. Because entering your password and username in config.yml, it looks like your user password is ignored.

**Comment by ChadSikorra** [comment]
Partially resolved with https://github.com/ldaptools/ldaptools-bundle/commit/05b46e81feacc2847c8a9297b23286003c0f82d1. That was the easy part anyway. This makes it possible to not need a username/password for the domain configurations. However, this only helps if you're using this bundle for the authentication piece, and not the user provider piece too.

Still thinking about how to handle the rest.

**Comment by ChadSikorra** [comment]
Ok, this is resolved now with this addition too https://github.com/ldaptools/ldaptools-bundle/commit/396fc40758f8180333703519bb6e692c3f59a8ac. Now all that is technically required for a domain config is a domain name.  It will still work for LDAP authentication and the the LDAP user provider. This is currently Guard only. I still need to look into the other LdapAuthenticationProvider.

Also, this currently requires the master branch of LdapTools. As I had to make a change to the requirements of the DomainConfiguration class. I will need to tag a new version there and then bump the version here.

**Comment by ChadSikorra** [comment]
Closing. I have bumped the required LdapTools version (which was needed for this), and both the LDAP Guard and Authentication Provider support using user supplied credentials for the user provider (if using this bundles LdapUserProvider). So now the domain config username/password is truly optional (depending on your needs and how you want to configure this anyway).

I have tagged a new version of this bundle. Let me know if you have any issues.
